### PR TITLE
Add options to select a signature when converting from TensorFlow Hub.

### PR DESCRIPTION
**Pull request by vbardiovskyg**
This change is [Reviewable](https://reviewable.io/reviews/tensorflow/tfjs-converter/107)

### CODE REVIEW

**Comment by pyu10055** [comment]
:lgtm_strong: Thanks couple minor comments.

---

Reviewed 4 of 4 files at r1.
Review status: all files reviewed at latest revision, all discussions resolved, some commit checks failed.

---

*[README.md, line 80 at r1](https://reviewable.io/reviews/tensorflow/tfjs-converter/107#-LBgH5F7EcU2ME5Aui5I:-LBgH5F7EcU2ME5Aui5J:b-7srvcu) ([raw file](https://github.com/tensorflow/tfjs-converter/blob/b954526ff7062a8364934f813219b4a97f854da0/README.md#L80)):*
> ```
> tf_hub_signature
> ```

---

*[README.md, line 80 at r1](https://reviewable.io/reviews/tensorflow/tfjs-converter/107#-LBgH5IR-iiYVqdQUxTZ:-LBgH5IR-iiYVqdQUxT_:brx37yt) ([raw file](https://github.com/tensorflow/tfjs-converter/blob/b954526ff7062a8364934f813219b4a97f854da0/README.md#L80)):*
> ```Markdown
> |`--output_node_names`| The names of the output nodes, separated by commas.|
> |`--saved_model_tags` | Only applicable to SavedModel conversion, Tags of the MetaGraphDef to load, in comma separated format. Defaults to `serve`.|
> |`--tf_hub_signature` | Only applicable to TensorFlow Hub module conversion, signature to load. Defaults to `default`. See https://www.tensorflow.org/hub/common_signatures/.|
> ```

can we name this something more generic, I think it could be used for saved model as well.
maybe signature_name ?

---

*[python/tensorflowjs/converters/tf_saved_model_conversion.py, line 297 at r1](https://reviewable.io/reviews/tensorflow/tfjs-converter/107#-LBgHmFLAyQ5XEvV1hTm:-LBgHmFLAyQ5XEvV1hTn:b-95540h) ([raw file](https://github.com/tensorflow/tfjs-converter/blob/b954526ff7062a8364934f813219b4a97f854da0/python/tensorflowjs/converters/tf_saved_model_conversion.py#L297)):*
> ```
> signature
> ```
should the signature have a default value? 

---


*Comments from [Reviewable](https://reviewable.io/reviews/tensorflow/tfjs-converter/107#-:-LBgIV5BD9tT77-MQ2hv:b-qds30t)*

**Comment by vbardiovskyg** [comment]
Review status: 1 of 4 files reviewed at latest revision, 3 unresolved discussions, some commit checks failed.

---

*[README.md, line 80 at r1](https://reviewable.io/reviews/tensorflow/tfjs-converter/107#-LBgH5F7EcU2ME5Aui5I:-LBtdY3x1pZ2GMxZLH2H:b-896fix) ([raw file](https://github.com/tensorflow/tfjs-converter/blob/b954526ff7062a8364934f813219b4a97f854da0/README.md#L80)):*
<details><summary><i>Previously, pyu10055 (Ping Yu) wrote…</i></summary><blockquote>

> ```
> tf_hub_signature
> ```

</blockquote></details>

Done.

---

*[README.md, line 80 at r1](https://reviewable.io/reviews/tensorflow/tfjs-converter/107#-LBgH5IR-iiYVqdQUxTZ:-LBtdYONA-adTtkSqUwc:b-hil5zl) ([raw file](https://github.com/tensorflow/tfjs-converter/blob/b954526ff7062a8364934f813219b4a97f854da0/README.md#L80)):*
<details><summary><i>Previously, pyu10055 (Ping Yu) wrote…</i></summary><blockquote>

can we name this something more generic, I think it could be used for saved model as well.
maybe signature_name ?
</blockquote></details>

Done. I kept the original description, it can be changed when new functionality is added.

---

*[python/tensorflowjs/converters/tf_saved_model_conversion.py, line 297 at r1](https://reviewable.io/reviews/tensorflow/tfjs-converter/107#-LBgHmFLAyQ5XEvV1hTm:-LBtdxYrEdOBDZINlzSW:b-896fix) ([raw file](https://github.com/tensorflow/tfjs-converter/blob/b954526ff7062a8364934f813219b4a97f854da0/python/tensorflowjs/converters/tf_saved_model_conversion.py#L297)):*
<details><summary><i>Previously, pyu10055 (Ping Yu) wrote…</i></summary><blockquote>

> ```
> signature
> ```
should the signature have a default value? 
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io/reviews/tensorflow/tfjs-converter/107)*

**Comment by vbardiovskyg** [comment]
Also fixed a flag typo (https://github.com/tensorflow/tfjs-converter/pull/112).

**Comment by caisq** [comment]
Thanks for fixing the bugs. I  think we should do a patch release (0.3.1) after the fixes gets in (together with https://github.com/tensorflow/tfjs-converter/pull/112)

@vbardiovskyg It would be great if we can add a test of the tf_hub conversion to test_pip_package.py, but given it downloads quite a lot of data from the internet, I'm not sure that test would be very reliable or efficient. Is there a smaller model on tf_hub that can be used for testing?

---

Review status: 1 of 4 files reviewed at latest revision, 3 unresolved discussions, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io/reviews/tensorflow/tfjs-converter/107#-:-LC-1nTz7rB0a2sMFPbi:bvfeh1i)*

**Comment by vbardiovskyg** [comment]
Hi, we are already testing with a module created on fly in: https://github.com/tensorflow/tfjs-converter/blob/master/python/tensorflowjs/converters/tf_saved_model_conversion_test.py#L248

---

Review status: 1 of 4 files reviewed at latest revision, 3 unresolved discussions, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io/reviews/tensorflow/tfjs-converter/107#-:-LC-32SrErfrUmfEINCm:bw7a7zm)*

**Comment by vbardiovskyg** [comment]
I can add a test to test_pip_package but wondering whether we can avoid duplication of the create_hub_module function.

---

Review status: 1 of 4 files reviewed at latest revision, 3 unresolved discussions, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io/reviews/tensorflow/tfjs-converter/107#-:-LC-3cXjDG5tXjQImsF2:biuzet)*

**Comment by caisq** [comment]
Yep, that test is good. But it doesn't cover the packaging of the functionality at the level of the pip binary. It didn't catch the typo, for example. Adding such a new test in test_pip_package.py would be valuable.

---

Review status: 1 of 4 files reviewed at latest revision, 3 unresolved discussions, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io/reviews/tensorflow/tfjs-converter/107#-:-LC-3jnC1xPExzqxX49W:b36w6t7)*

**Comment by caisq** [comment]
Review status: 1 of 5 files reviewed at latest revision, 3 unresolved discussions, all commit checks successful.

---

*[README.md, line 77 at r5](https://reviewable.io/reviews/tensorflow/tfjs-converter/107#-LC-9Pdp3WYEo5uPZhEI:-LC-9Pdp3WYEo5uPZhEJ:bt7j0bb) ([raw file](https://github.com/tensorflow/tfjs-converter/blob/489fafa666a1637be71ad0487dee3e5f6a4d88b2/README.md#L77)):*
> ```
> tf_hub
> ```

I know it's unrelated to this PR. But can you add 'keras for Keras HDF5' here? 

---

*[python/tensorflowjs/converters/converter.py, line 156 at r5](https://reviewable.io/reviews/tensorflow/tfjs-converter/107#-LC-9tIlAyND_WAwzhJF:-LC-9tIlAyND_WAwzhJG:bah54se) ([raw file](https://github.com/tensorflow/tfjs-converter/blob/489fafa666a1637be71ad0487dee3e5f6a4d88b2/python/tensorflowjs/converters/converter.py#L156)):*
> ```
> '--signature_name'
> ```

Can you add a check for if `--signature_name` is specified and `--input_format` is not `tf_hub`, in which case an error should be thrown?

---


*Comments from [Reviewable](https://reviewable.io/reviews/tensorflow/tfjs-converter/107)*

**Comment by pyu10055** [comment]
Reviewed 1 of 3 files at r2, 1 of 1 files at r4, 2 of 2 files at r5.
Review status: all files reviewed at latest revision, 5 unresolved discussions, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io/reviews/tensorflow/tfjs-converter/107)*

**Comment by vbardiovskyg** [comment]
Review status: 1 of 5 files reviewed at latest revision, 5 unresolved discussions, all commit checks successful.

---

*[README.md, line 77 at r5](https://reviewable.io/reviews/tensorflow/tfjs-converter/107#-LC-9Pdp3WYEo5uPZhEI:-LC-QG4jCiv1_Dn71zcC:b-896fix) ([raw file](https://github.com/tensorflow/tfjs-converter/blob/489fafa666a1637be71ad0487dee3e5f6a4d88b2/README.md#L77)):*
<details><summary><i>Previously, caisq (Shanqing Cai) wrote…</i></summary><blockquote>

> ```
> tf_hub
> ```

I know it's unrelated to this PR. But can you add 'keras for Keras HDF5' here? 
</blockquote></details>

Done.

---

*[python/tensorflowjs/converters/converter.py, line 157 at r5](https://reviewable.io/reviews/tensorflow/tfjs-converter/107#-LC-UWGy2r-LPK_gY1xs:-LC-UWGy2r-LPK_gY1xt:b-ufb478) ([raw file](https://github.com/tensorflow/tfjs-converter/blob/489fafa666a1637be71ad0487dee3e5f6a4d88b2/python/tensorflowjs/converters/converter.py#L157)):*
> ```Python
>   parser.add_argument(
>       '--signature_name',
>       type=str,
> ```

There is actually a default for this value so it will always be set. The advantage is a cleaner API. If you insist on the change, please advise how to to differentiate 

---


*Comments from [Reviewable](https://reviewable.io/reviews/tensorflow/tfjs-converter/107)*

**Comment by vbardiovskyg** [comment]
Review status: 1 of 5 files reviewed at latest revision, 6 unresolved discussions, all commit checks successful.

---

*[python/tensorflowjs/converters/converter.py, line 157 at r5](https://reviewable.io/reviews/tensorflow/tfjs-converter/107#-LC-UWGy2r-LPK_gY1xs:-LC2rwa8-BlUOFXjm19b:b2tifny) ([raw file](https://github.com/tensorflow/tfjs-converter/blob/489fafa666a1637be71ad0487dee3e5f6a4d88b2/python/tensorflowjs/converters/converter.py#L157)):*
<details><summary><i>Previously, vbardiovskyg wrote…</i></summary><blockquote>

There is actually a default for this value so it will always be set. The advantage is a cleaner API. If you insist on the change, please advise how to to differentiate 
</blockquote></details>

Unfinished draft comment, anyway I went with no default value in the end.

---


*Comments from [Reviewable](https://reviewable.io/reviews/tensorflow/tfjs-converter/107)*

**Comment by caisq** [comment]
:lgtm_strong:

---

Review status: 1 of 5 files reviewed at latest revision, 6 unresolved discussions, all commit checks successful.

---

*[python/tensorflowjs/converters/converter.py, line 208 at r7](https://reviewable.io/reviews/tensorflow/tfjs-converter/107#-LC44El54BqCMmIZuT7j:-LC44El54BqCMmIZuT7k:bp3209x) ([raw file](https://github.com/tensorflow/tfjs-converter/blob/fc5ff8d69516683eee0895e08caecf9e32bb4b9d/python/tensorflowjs/converters/converter.py#L208)):*
> ```
> (F
> ```

nit: no parentheses are required here as per Google Python style.

---


*Comments from [Reviewable](https://reviewable.io/reviews/tensorflow/tfjs-converter/107#-:-LC44M5p1IDKmg88aBpb:b-jbsf8z)*

**Comment by vbardiovskyg** [comment]
Review status: 1 of 5 files reviewed at latest revision, 7 unresolved discussions, some commit checks pending.

---

*[python/tensorflowjs/converters/converter.py, line 208 at r7](https://reviewable.io/reviews/tensorflow/tfjs-converter/107#-LC44El54BqCMmIZuT7j:-LC4IBGB11zowyevZ19w:b-896fix) ([raw file](https://github.com/tensorflow/tfjs-converter/blob/fc5ff8d69516683eee0895e08caecf9e32bb4b9d/python/tensorflowjs/converters/converter.py#L208)):*
<details><summary><i>Previously, caisq (Shanqing Cai) wrote…</i></summary><blockquote>

> ```
> (F
> ```

nit: no parentheses are required here as per Google Python style.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io/reviews/tensorflow/tfjs-converter/107)*
